Project3/Part_A/main.cpp

```diff
@@ -17,30 +21,28 @@ void findMatches(Dictionary dict, Grid grid)
   int width = grid.getWidth(), height = grid.getHeight();
   for (int row = 0; row < height; row++) {
     for (int col = 0; col < width; col++) {
-      std::vector<std::string> found;
-      search(grid, row, col, found);
-
-      for (const std::string &match : found) {
-        if (dict.lookupWord(match)) {std::cout << match << std::endl;}
-      }
+      searchDirections(dict, grid, row, col);
     }
   }
 }
 
-void search(Grid grid, int row, int col, std::vector<std::string> found)
+void searchDirections(Dictionary dict, Grid grid, int row, int col)
 {
-  int x[] = {};
-  int y[] = {};
+  dict.sortWords();
+  // directional arrays
+  int x[] = {-1, -1, -1, 0, 0, 1, 1, 1};
+  int y[] = {-1, 0, 1, -1, 1, -1, 0, 1};
 
   for (int direction = 0; direction < 8; direction++) {
     std::string current = "";
-    int rd = row + x[direction];
-    int cd = col + y[direction];
+    int rd = row;
+    int cd = col;
 
-    while (rd > -1 && rd < grid.getHeight() && cd > -1 && grid.getWidth()) {
+    while (rd > -1 && rd < grid.getHeight() && cd > -1 && cd < grid.getWidth()) {
```

Comment:
This doesn't allow for words to wrap around, if a dictionary change to allow for getting max word length would help then one of us can do that